src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageUtility.java

```diff
@@ -0,0 +1,110 @@
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import ca.corefacility.bioinformatics.irida.exceptions.ConcatenateException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Interface describing methods for performing storage actions
+ */
+
+public interface IridaFileStorageUtility {
+	//Valid extensions to try to concatenate with this tool
+	public static final List<String> VALID_EXTENSIONS = Lists.newArrayList("fastq", "fastq.gz");
+	/**
+	 * Get a temporarry file from storage
+	 *
+	 * @param file The {@link Path} to the file
+	 * @return {@link File} which was retrieved from path
+	 */
+	public File getTemporaryFile(Path file);
+
+	/**
+	 * Get file size in bytes
+	 *
+	 * @param file The {@link Path} to the file
+	 * @return {@link Long} size of file retrieved from path
+	 */
+	public Long getFileSize(Path file);
+
+	/**
+	 * Write file to storage (azure, aws, or local)
+	 *
+	 * @param source The {@link Path} to the file
+	 * @param target The {@link Path} to where file should be moved
+	 * @param sequenceFileDir The {@link Path} to sequence file directory
+	 * @param sequenceFileDirWithRevision The {@link Path} to sequence file revision directory
+	 */
+	public void writeFile(Path source, Path target, Path sequenceFileDir, Path sequenceFileDirWithRevision);
+
+	/**
+	 * Returns if the storage type is local or not
+	 *
+	 * @return {@link Boolean#TRUE} if local, {@link Boolean#FALSE} if not.
+	 */
+	public boolean storageTypeIsLocal();
+
+	/**
+	 * Gets the file name from the storage type that the file
+	 * is saved to (azure, aws, or local disk)
+	 *
+	 * @param file The path to the file for which to get name for
+	 * @return {@link String} The file name for the file
+	 */
+	public String getFileName(Path file);
+
+
+	/**
+	 * Checks if file exists
+	 *
+	 * @param file The path to the file
+	 * @return true if file exists otherwise false
+	 *
+	 */
+	public boolean fileExists(Path file);
+
+	/**
+	 * Gets the file inputstream
+	 *
+	 * @param file The path to the file
+	 * @return file inputstream
+	 *
+	 */
+	public InputStream getFileInputStream(Path file);
+
+	/**
+	 * Checks if file is gzipped
+	 *
+	 * @param file The path to the file
+	 * @return true if file is gzipped otherwise false
+	 * @throws IOException if file can't be read
+	 *
+	 */
+	public boolean isGzipped(Path file) throws IOException;
+
+	/**
+	 * Append a {@link SequenceFile} to a {@link Path} on the filesystem
+	 *
+	 * @param target the {@link Path} to append to
+	 * @param file   the {@link SequenceFile} to append to the path
+	 * @throws ConcatenateException if there is an error appending the file
+	 */
+	public void appendToFile(Path target, SequenceFile file) throws ConcatenateException;
+
+	/**
+	 * Get the extension of the files to concatenate
```

Comment:
See comment for `public String getFileExtension` below

Reply:
Ok that's fine.  I think the comment here should be updated to remove "to concatenate".  It's a bit confusing since this isn't specifically an concatenation thing.

Reply:
Yeah I totally agree after looking at it. Will update it. Thanks!

Reply:
Updated in 660b10c024aff6e9ee61bcf60eb6f05820f8fa01